src/Microsoft.SqlTools.ServiceLayer/LanguageServices/ConnectedBindingQueue.cs

```diff
@@ -72,12 +72,24 @@ public ConnectedBindingQueue(bool needsMetadata)
         private string GetConnectionContextKey(ConnectionInfo connInfo)
         {
             ConnectionDetails details = connInfo.ConnectionDetails;
-            return string.Format("{0}_{1}_{2}_{3}",
+            string key = string.Format("{0}_{1}_{2}_{3}",
                 details.ServerName ?? "NULL",
                 details.DatabaseName ?? "NULL",
                 details.UserName ?? "NULL",
                 details.AuthenticationType ?? "NULL"
             );
+
+            if (!string.IsNullOrEmpty(details.DatabaseDisplayName))
```

Comment:
in what scenario we need to add the database display name which is different than database name?

Reply:
this is to differentiate connections to "default" (no db name provided) and the default DB (when 'master') has been provided.  Currently if <default> and 'master' connections are in the same group they will have the same connection key, and bad things will happen.